tests/test_throttler.py

```diff
@@ -26,6 +27,56 @@ def f2() -> str:
     assert mock2.call_count == 10
 
 
+def test_debouncer_method(qtbot):
+    class A(QObject):
+        def __init__(self):
+            super().__init__()
+            self.count = 0
+
+        def callback(self):
+            self.count += 1
+
+    a = A()
+    assert all(not isinstance(x, ThrottledCallable) for x in a.children())
+    b = qdebounced(a.callback, timeout=4)
+    assert any(isinstance(x, ThrottledCallable) for x in a.children())
+    for _ in range(10):
+        b()
+
+    qtbot.wait(5)
+
+    assert a.count == 1
+
+
+def test_debouncer_method_definition(qtbot):
+    mock1 = Mock()
+
+    class A(QObject):
+        def __init__(self):
+            super().__init__()
+            self.count = 0
+
+        @qdebounced(timeout=4)
+        def callback(self):
+            self.count += 1
+
+        @qdebounced(timeout=4)
+        @staticmethod
```

Comment:
```suggestion
        @staticmethod
        @qdebounced(timeout=4)
```

i think this might fix the test error

Reply:
(or at least, if it doesn't ... then it shows a good thing to test)

Reply:
alternatively, since this doesn't *look* like it should have been an error on python3.8/3.9, you could try to figure out why staticmethods weren't `callable` back then, and special case it in get_max_args

Reply:
yeah, I'm playing with it too... there's definitely something stange happening.  even if I pull out the actual function from the static method at `func.__func__`, its *still* getting reassigned back to `__wrapped__` at some later point, and causing an error (on py38)

Reply:
Could you paste the code snippet that reproduces the problem?

Reply:
i'm just running your `test_debouncer_method_definition` test.  i think i have a fix... just a min

Reply:
oh I see you posted a fix as well... hang on

Reply:
But how it pass on CI and fail on your machine?

Reply:
the issue, by the way, was that `__set_name__` appears to be *reassigning* `self.__wrapped__`.

>But how it pass on CI and fail on your machine?

because I hadn't pulled your new variant with a new class